### PR TITLE
Implement connect payment page for Adyen

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Implement a second connect payment page using Adyen

- Support DK/NO users

- Re-use some utilities from the first connect payment page

- Now we can use this link in HOPE: `/api/auth/exchange/{AUTHORIZATION_CODE}?next=/no/payment/connect-legacy`

## Justify why they are needed

- We want to support DK/NO users update their payment method in the new platform

- The design is a bit ad-hoc, but it's a quick solution to support DK/NO users

- We can't use any newer version of the Adyen SDK because it's not compatible with the current version of the Adyen API (v66 on the backend)

- I decided the keep translations hardcoded in `useAdyenTranslations` since it's the only ones we need for DK/NO

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2832** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2832" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2832?utm_source=stack-comment).

File: apps/store/next-csp.config.js [hunk 1]
```diff
@@ -146,6 +157,10 @@ const frameSrc = [
   // Insurely
   'https://blocks.insurely.com',
 
+  // Adyen
+  ...adyenOrigins,
+  'https://pay.google.com',
```
Comment (alebedev) [inline apps/store/next-csp.config.js:162]: Let's try to tell Adyen we're not interested in Google Pay when we render payment form

File: apps/store/src/pages/api/adyen-callback/[locale].ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  try {
+    const apolloClient = await initializeApolloServerSide({ req, res })
+    await submitAdyenRedirection(apolloClient, { md, pares })
+  } catch (error) {
+    console.warn('Payment Connect Legacy | Error submitting redirection: ', error)
+    res.redirect(307, PageLink.paymentConnectLegacyError({ locale: req.query.locale }))
+    return
+  }
+
+  res.redirect(307, PageLink.paymentConnectLegacySuccess({ locale: req.query.locale }))
```
Comment (alebedev) [inline apps/store/src/pages/api/adyen-callback/[locale].ts:23]: Should be 302, right?

File: apps/store/src/services/adyen/useGetAdyenCheckout.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        allowPaymentMethods: [
+          // Credit Cards
+          'scheme',
+          'applepay',
+        ],
+
+        paymentMethodsConfiguration: {
+          ...paymentMethodsConfiguration,
+
+          applepay: {
```
Comment (alebedev) [inline apps/store/src/services/adyen/useGetAdyenCheckout.ts:37]: would be logical to put it into `paymentMethodsConfiguration`, no?

File: apps/store/src/services/adyen/useGetAdyenCheckout.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        })
+        element.setStatus('error')
+        return
+      }
+
+      if (data.__typename === 'TokenizationResponseAction') {
+        datadogLogs.logger.info('Adyen Submit | Action')
+        const action = JSON.parse(data.action) as PaymentAction
+        element.handleAction(action)
+      } else {
```
Comment (alebedev) [inline apps/store/src/services/adyen/useGetAdyenCheckout.ts:99]: Nit: would be easier to read if flattened to `if / else if / else`

**Review by alebedev — APPROVED** [review]
Very good!  Few minor comments

**Review by guilhermespopolin — APPROVED** [review]
Looking good :)

**Pull request merged**